pkg/clusterrouter/proxy/proxy_server.go

```diff
@@ -58,6 +65,10 @@ type authGetter func() (string, error)
 
 type transportGetter func() (http.RoundTripper, error)
 
+type tokenCreator func(context.Context, ClusterContextGetter, string, string) (string, error)
+
+type impersonatorAccountTokenGetter func(user.Info, ClusterContextGetter, string) (string, error)
+
 type errorResponder struct {
 }
```

Comment:
nit: Maybe we can put all type declarations in a type block? Like
```go
type (
	urlGetter       func() (url.URL, error)
	authGetter      func() (string, error)
	transportGetter func() (http.RoundTripper, error)
	errorResponder  struct{}
)
```

Also `authGetter` doesn't seem to be used.